api-gateway/src/main/resources/static/scripts/visits/visits.controller.js

```diff
@@ -13,8 +13,30 @@ angular.module('visits')
 
         $http.get("api/gateway/visits/"+petId).then(function (resp) {
             self.visits = resp.data;
+            self.sortFetchedVisits();
         });
 
+        self.sortFetchedVisits = function() {
+            let dateObj = new Date();
+            var dd = String(dateObj.getDate()).padStart(2, '0');
+            var mm = String(dateObj.getMonth() + 1).padStart(2, '0');
+            var yyyy = dateObj.getFullYear();
+            let currentDate = Date.parse(yyyy + '-' + mm + '-' + dd);
+
+            self.upcomingVisits = [];
+            self.previousVisits = [];
+
+            $.each(self.visits, function(i, visit) {
+                let selectedVisitDate = Date.parse(visit.date);
+
+                if(selectedVisitDate >= currentDate) {
+                    self.upcomingVisits.push(visit);
+                } else {
+                    self.previousVisits.push(visit);
+                }
+            });
+        }
+
```

Comment:
The point of my back-end story (fetch previous and scheduled visits separately) was so that you didn't have to do this logic in the front-end.

Reply:
For sure, I'll refactor the code to use our service's methods once we get the mappings for them in API.
It should be better to do it this way as well since there's no sorting needing to be done and the visits can be displayed ASAP, I believe.

Reply:
Ohhhh you're right I forgot that we had to go through the API